src/components/pacbio/PacbioTagEditModal.vue

```diff
@@ -0,0 +1,102 @@
+<template>
+  <div>
+    <b-button :id="'editTag'+this.tag.id" size="sm" @click="show" variant="outline-primary">Edit</b-button>
+
+    <b-modal
+      id="editTagModal"
+      ref="modal"
+      title="Edit Tag"      
+    >    
+
+      <b-form id="editTagForm" v-on:submit.prevent>
+        <b-form-group id="input-group-1"
+                      label="Sample tag: "
+                      label-for="input-1">
+                      <span>{{ this.tag.sample_name }}</span>
+          <b-form-select ref="tagSelection" 
+                        id="tagSelection" 
+                        v-model="selectedSampleTagId"
+                        :options="tags">
+          </b-form-select>
+        </b-form-group>
+      </b-form>
+
+      <template v-slot:modal-footer="{ ok, cancel }">
+        <b-button @click="cancel()">
+          Cancel
+        </b-button>
+
+        <b-button variant="success" @click="update()">
+          Update Tag
+        </b-button>
+      </template>
+
+    </b-modal>
+  </div>
+</template>
+
+<script>
+
+import ModalHelper from '@/mixins/ModalHelper'
+import { mapActions, mapGetters } from 'vuex'
+
+export default {
+  name: 'PacbioLibraryTagModal',
+  mixins: [ModalHelper],
+  data() {
+    return {
+      tags: [],
+      selectedSampleTagId: ""
+    }
+  },
+  props: {
+    tag: {
+      type: [Object]
+    }
+  },
+  methods: {
+    async update() {
+      try {
+        let payload = {selectedSampleTagId: this.selectedSampleTagId, request_library_id: this.tag.id }
+        await this.updateTag(payload)
+        this.alert('Tag updated', 'success')
+      } catch (err) {
+        this.alert('Failed to update Tag. ' + err, 'danger')
+      }
+      this.hide()
+      this.$emit('reloadPage')
```

Comment:
Hm this makes me wonder that something isn't quite right if you have to manually reload the page - might be something in the VueX store asyn reloading. Will have a proper look after lunch